### DISCUSSION TITLE
GPU is not used during prediction ?

### DISCUSSION THREAD

**Discussion by patcharees** [obss/sahi#908] [Q&A]
Hi,

I am using sahi with yolov8 segmentation (https://github.com/obss/sahi/pull/882). I created Yolov8DetectionModel with gpu as:

```
detection_model = Yolov8DetectionModel(
            model_path=yolov8_model_path,
            confidence_threshold=CONFIDENCE_THRESHOLD,
            device='cuda:0',
            category_remapping=None,
            load_at_init=True,
            image_size=IMAGE_SIZE,
    )

```
But when I run my script to predict, I noticed from nvidia-smi that gpu utilization is very low (< 10%)

My problem is when I use high overlap and number of slices is high, then my script, which loop through all images to predict one image at a time, is very slow. In addition it is killed from Out of memory after it predicts a few images. 

My image is large 8192 * 5460

Any suggestions to improve speed and to avoid OOM?

**Comment by jokober** [comment]
Can you send your nvidia-smi output? How much normal RAM do you have?

Have you studied the output? I'm asking because fallback to GPU and its reason is often described in the output.

Do you use get_sliced_prediction or (batch) predict? (https://github.com/obss/sahi/blob/main/demo/inference_for_yolov8.ipynb)

**Comment by patcharees** [comment]
Hi @jokober 

I used get_sliced_prediction (not batch). My pc has 62 Gi Total RAM. What does it mean to study the output? Currently Sahi prints out only 'Performing prediction on 216 number of slices.' during processing, how can I get more info?

A snapshot immediately after the python script was killed:

![OOM](https://github.com/obss/sahi/assets/104826856/33fcef2a-1212-4ba7-afbb-64b7e405a974)

A snapshot of nvidia-smi during the python script was running:

![nvidia-smi](https://github.com/obss/sahi/assets/104826856/475f9092-9c32-404a-8fe8-79d01d5c4ae4)

**Comment by patcharees** [comment]
Btw I found out this setting merge_buffer_length = 100 that can avoid OOM. But still the prediction is very slow.

**Comment by jokober** [comment]
Using batch prediction (predict() function) will likely improve speed as it will utilize your gpu more.

**Comment by pkcktkksh98** [comment]
@patcharees hi, i encounter the same problem as you did. Did you manage to use GPU?

**Comment by patcharees** [comment]
@pkcktkksh98 sorry for late reply. Unfortunately I still can not manage to use GPU. You?

I tried batch prediction (predict() function) as suggested above. But it did not help. I monitored the GPU usage while prediction, it was under 10% all the time.

**Reply by pkcktkksh98** [reply]
Same goes here. Still couldnt inference it using the GPU.
